File: typescript/src/types/team.ts
import { DateTime, UUID } from './common';

/**
 * Team member response object
 */
export interface TeamMember {
  id: UUID;
  name: string;
  designation: string | null;
  bio: string | null;
  about: string | null;
  avatar_url: string | null;
  avatar: string | null;
  email: string | null;
  phone: string | null;
  social_links: any | null;
  category: string | null;
  sort_order: number | null;
  active: boolean;
  created_at: DateTime;
  updated_at: DateTime;
}

/**
 * Create team member request body
 */
export interface CreateTeamMemberBody {
  name: string;
  designation?: string | null;
  bio?: string | null;
  about?: string | null;
  avatarUrl?: string | null;
  avatar_url?: string | null;
  email?: string | null;
  phone?: string | null;
  socialLinks?: any | null;
  social_links?: any | null;
  category?: string | null;
  sortOrder?: number | null;
  sort_order?: number | null;
  active?: boolean;
}

/**
 * Update team member request body
 */
export interface UpdateTeamMemberBody {
  name?: string;
  designation?: string | null;
  bio?: string | null;
  about?: string | null;
  avatarUrl?: string | null;
  avatar_url?: string | null;
  email?: string | null;
  phone?: string | null;
  socialLinks?: any | null;
  social_links?: any | null;
  category?: string | null;
  sortOrder?: number | null;
  sort_order?: number | null;
  active?: boolean;
}

/**
 * Team member list query parameters
 */
export interface TeamMemberListParams {
  category?: string;
  active?: boolean;
  search?: string;
  limit?: number;
  page?: number;
}

/**
 * Team member statistics response
 */
export interface TeamMemberStats {
  total: number;
  active: number;
  inactive: number;
}

File: python/sakha_cms/models/team.py
"""Team module models."""

from pydantic import BaseModel, ConfigDict, Field
from datetime import datetime
from typing import Optional


class TeamMember(BaseModel):
    """Team member."""

    model_config = ConfigDict(populate_by_name=True)

    id: str
    project_id: Optional[str] = Field(None, alias="project_id")
    name: str
    email: str
    phone: Optional[str] = None
    role: str  # admin|editor|viewer
    avatar_url: Optional[str] = Field(None, alias="avatar_url")
    is_active: bool = Field(default=True, alias="is_active")
    joined_at: datetime = Field(alias="joined_at")
    created_at: datetime = Field(alias="created_at")
    updated_at: datetime = Field(alias="updated_at")


class CreateTeamMemberBody(BaseModel):
    """Create team member request."""

    model_config = ConfigDict(populate_by_name=True)

    name: str
    email: str
    phone: Optional[str] = None
    role: str = "viewer"
    avatar_url: Optional[str] = Field(None, alias="avatarUrl")


class UpdateTeamMemberBody(BaseModel):
    """Update team member request."""

    model_config = ConfigDict(populate_by_name=True)

    name: Optional[str] = None
    phone: Optional[str] = None
    role: Optional[str] = None
    avatar_url: Optional[str] = Field(None, alias="avatarUrl")
    is_active: Optional[bool] = Field(None, alias="isActive")


class TeamStats(BaseModel):
    """Team statistics."""

    model_config = ConfigDict(populate_by_name=True)

    total_members: int = Field(alias="totalMembers")
    active_members: int = Field(alias="activeMembers")
    admin_members: int = Field(alias="adminMembers")
    editor_members: int = Field(alias="editorMembers")
    viewer_members: int = Field(alias="viewerMembers")
